homeassistant/components/logger/__init__.py

```diff
@@ -103,6 +111,15 @@ def _set_log_level(logger, level):
     getattr(logger, "orig_setLevel", logger.setLevel)(LOGSEVERITY[level])
 
 
+def _add_log_filter(logger, patterns):
+    """Add a Filter to the logger based on a regexp of the filter_str."""
+
+    def filter_func(logrecord):
+        return not any(re.match(p, logrecord.getMessage()) for p in patterns)
```

Comment:
```suggestion
        return not any(p.match(logrecord.getMessage()) for p in patterns)
```

When `p` is precompiled with `re.compile` use `p.match()`

Reply:
Amending, since this (neatly) removes the dependency on `import re`